### PR TITLE
Create assignproj.yml

### PR DESCRIPTION
**Pull request by angelatan2**
Auto assigns project to an issue on create

_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-ruby-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/master/CODE_OF_CONDUCT.md)._

# Overview
Add GHA to automatically associate the issue to the correct engineering project.

# Related Github Issue
Include a link to the related GitHub issue, if applicable

# Testing
Need to add SECRET with a TOKEN to pass.
1) Create a new issue.
2) Verify that it is automatically assigned to the correct Ruby engineering project and it shows up on Triage Column.

### CODE REVIEW

File: .github/workflows/assignproj.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Auto Assign to Project
+
+on:
+  issues:
+    types: [opened]
+  pull_request:
+    types: [opened]
```
Comment (rachelleahklein) [inline .github/workflows/assignproj.yml:7]: I am guessing we also want to run this on reopened issues and PRs, correct?

File: .github/workflows/assignproj.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  MY_GITHUB_TOKEN: ${{ secrets.RUBY_GITHUB_TOKEN }}
+
+jobs:
+  assign_one_project:
+    runs-on: ubuntu-latest
+    name: Assign to One Project
+    steps:
+    - name: Assign NEW issues and NEW pull requests to project 2
+      uses: srggrs/assign-one-project-github-action@1.2.0
+      if: github.event.action == 'opened'
```
Comment (rachelleahklein) [inline .github/workflows/assignproj.yml:18]: My earlier comment applies here as well.

**Review by rachelleahklein — COMMENTED** [review]
@angelatan2, thank you for submitting a PR to automate this work! I have one comment on the setup. Also, the secret has already been added, correct?

File: .github/workflows/assignproj.yml (hunk 1 above)
Comment (angelatan2) [reply .github/workflows/assignproj.yml:7]: Hi Rachel, once an issue ticket is assigned a projects the first time, we don't have to reassigned it on a reopened.  Is there a different use case that I'm not following.  Moving this conversation to slack.

Comment (rachelleahklein) [reply .github/workflows/assignproj.yml:7]: I was thinking about the (admittedly not all that likely) possibility that an issue or pull request created before this PR was merged would be reopened and left without a project assigned. Per our Slack conversation, @angelatan2 will run a query separately to make sure there are no old ones without a project name.

**Review by rachelleahklein — APPROVED** [review]
Thank you for automating away toil for the team, @angelatan2!

**Pull request merged**